libraries/Microsoft.Bot.Builder.Dialogs/DialogContextState.cs

```diff
@@ -170,7 +170,19 @@ public T GetValue<T>(string pathExpression, T defaultValue = default(T))
 
         public T GetValue<T>(object o, string pathExpression, T defaultValue = default(T))
         {
-            var result = JToken.FromObject(o).SelectToken(pathExpression);
+            JToken result = null;
+            if (o.GetType() == typeof(JArray))
```

Comment:
what if o == null? consider adding a null check and handling that case. can o be null? under what circumstances?